File: swift2.3学习/代码/day-2/6-block-oc/6-block-oc/ViewController.m
//
//  ViewController.m
//  6-block-oc
//
//  Created by gouzi on 2017/6/2.
//  Copyright © 2017年 wj. All rights reserved.
//

#import "ViewController.h"
#import "wjHttpTool.h"



@interface ViewController ()

@property (nonatomic, strong) wjHttpTool *tool;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.tool = [[wjHttpTool alloc] init];
}

// 网络请求
- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event {

    [self.tool wjLoadDataWithBlock:^(NSString *jsonData) {
        NSLog(@"view get data is : %@", jsonData);
    }];
    
}


- (void)dealloc {
    NSLog(@"dddddd");
}


@end

File: swift2.3学习/代码/day-2/8-懒加载/8-懒加载/ViewController.swift
//
//  ViewController.swift
//  8-懒加载
//
//  Created by gouzi on 2017/6/2.
//  Copyright © 2017年 wj. All rights reserved.
//

import UIKit

class ViewController: UIViewController {


    lazy var names : [String] = {
        print("======")
        return ["wj", "haha", "gouzi"]
    }()
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        
    }

    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        print(self.names)
        _ = names.count
    }

}

